src/components/ServiceList.tsx

```diff
@@ -14,14 +15,29 @@ function ServiceList() {
   const query = router.query;
   const searchQuery = query.search as string;
   const [view, setView] = useState(1);
-
+  const [isPopupVisible, setPopupVisible] = useState(false);
+  const [minRate, setMinRate] = useState<string>('');
```

Comment:
having all these useState is not really scalable and maintainable. I would recommend using a single state called filters and then we used this array and pass it directly to the backend